### PR TITLE
Implemented tests for Tax component

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
Unit tests covering the Tax component.

This is a split of #991

### CODE REVIEW

File: src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+class TaxTest extends AbstractEntityTest
+{
+    use Traits\IdentifiableTrait, Traits\EnabledTrait;
+
+    /**
+     * Sets up the fixture, for example, opens a network connection.
+     * This method is called before a test is executed.
+     */
+    protected function setUp()
```
Comment (mmoreram) [inline src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php]: Why this?

File: src/Elcodi/Component/Core/Tests/UnitTest/Entity/Traits/IdentifiableTrait.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Core\Tests\UnitTest\Entity\Traits;
+
+trait IdentifiableTrait
```
Comment (mmoreram) [inline src/Elcodi/Component/Core/Tests/UnitTest/Entity/Traits/IdentifiableTrait.php]: These traits are not used anymore. Not needed.
This is already tested automaticaly

File: src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                'type' => $this::GETTER_SETTER,
+                'getter' => 'getValue',
+                'setter' => 'setValue',
+                'value' => microtime(true),
+                'nullable' => false,
+            ],
+        ];
+    }
+
+    public function testToString()
```
Comment (mmoreram) [inline src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php]: We should "standarize" this as well.
Anyway, don't use `$this->object` and create here the mock.

File: src/Elcodi/Component/Tax/Tests/UnitTest/Factory/TaxFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Tax\Tests\UnitTest\Factory;
+
+use Elcodi\Component\Core\Tests\UnitTest\Factory\Abstracts\AbstractFactoryTest;
+
+class TaxFactoryTest extends AbstractFactoryTest
```
Comment (mmoreram) [inline src/Elcodi/Component/Tax/Tests/UnitTest/Factory/TaxFactoryTest.php]: phpdoc here, even if they are tests

File: src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php (hunk 1 above)
Comment (EmanueleMinotto) [inline src/Elcodi/Component/Tax/Tests/UnitTest/Entity/TaxTest.php]: I've used it to the the `__toString`, should it be included in the abstract system?

File: src/Elcodi/Component/Core/Tests/UnitTest/Entity/Traits/IdentifiableTrait.php (hunk 2 above)
Comment (EmanueleMinotto) [inline src/Elcodi/Component/Core/Tests/UnitTest/Entity/Traits/IdentifiableTrait.php]: Should I keep the `EnabledTrait`? There are also the `enable` and `disable` tests there.

Comment (mmoreram) [inline src/Elcodi/Component/Core/Tests/UnitTest/Entity/Traits/IdentifiableTrait.php]: All traits are tested already in the abstract class ;)

**Pull request merged**